app/services/AuditService.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import actionsmodel.AuthenticatedJourneyRequest
+import cats.syntax.eq._
+import com.github.ghik.silencer.silent
+import essttp.journey.model.Journey.Stages.ComputedTaxId
+import essttp.journey.model.Origin
+import essttp.journey.model.ttp.EligibilityCheckResult
+import models.audit.eligibility.{EligibilityCheck, EligibilityResult, EnrollmentReasons, TaxDetail}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class AuditService @Inject() (auditConnector: AuditConnector)(implicit ec: ExecutionContext) {
+
+  def auditEligibilityCheck(
+      journey:  ComputedTaxId,
+      response: EligibilityCheckResult
+  )(implicit r: AuthenticatedJourneyRequest[_], hc: HeaderCarrier): Unit = {
+    val auditEvent = toEligibilityCheck(journey, response)
+    auditConnector.sendExplicitAudit("EligibilityCheck", auditEvent)
+  }
+
+  // TODO: hook in when questions about business rules clarified
+  @silent
+  private def toEligibilityCheck(
+      journey:          ComputedTaxId,
+      enrollmentReason: Either[EnrollmentReasons.NotEnrolled, EnrollmentReasons.InactiveEnrollment]
+  )(implicit r: AuthenticatedJourneyRequest[_]): EligibilityCheck = {
+    EligibilityCheck(
+      EligibilityResult.Ineligible,
```

Comment:
named parameters might be helpful (and below)

Reply:
👍 added it [here](https://github.com/hmrc/essttp-frontend/pull/92/commits/2494869cf8e983d94983c15f96ec8a49d8ff40a2) looks clearer now I think